ultralytics/utils/metrics.py

```diff
@@ -738,8 +738,8 @@ def compute_ap(recall: list[float], precision: list[float]) -> tuple[float, np.n
         mrec (np.ndarray): Modified recall curve with sentinel values added at the beginning and end.
     """
     # Append sentinel values to beginning and end
-    mrec = np.concatenate(([0.0], recall, [1.0]))
-    mpre = np.concatenate(([1.0], precision, [0.0]))
+    mrec = np.concatenate(([0.0], recall, [recall[-1] if len(recall) > 0 else 1.0], [1.0]))
```

Comment:
❗ **CRITICAL**: Adding `recall[-1]` as another sentinel duplicates the last recall value, so `mrec` is no longer strictly increasing. `np.interp` (used a few lines below) requires strictly increasing `xp`; with this change we now hit a `ValueError: fp and xp are not of the same length` when `recall` is empty (producing `[0.0, 1.0, 1.0]`) and risk undefined interpolation results even when it isn’t. Please keep the end sentinel at `1.0` only.

**Suggested change:**
```suggestion
    mrec = np.concatenate(([0.0], recall, [1.0]))
```